src/ComposerCompatibleUpdater.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Tuf\ComposerIntegration;
+
+use Psr\Http\Message\StreamInterface;
+use Tuf\Client\Updater;
+use Tuf\Metadata\TargetsMetadata;
+
+/**
+ * Defines an updater that exposes additional information about TUF targets.
+ *
+ * This class is necessary because, although PHP-TUF normally is responsible for downloading
+ * and verifying its targets transparently (using its ::download() method), Composer doesn't
+ * really support overriding its HttpDownloader service -- at least, not without some seriously
+ * complicated and fragile sorcery. Therefore, this class exists specifically to allow
+ * Composer to download its packages and metadata how it wants, while still allowing us to
+ * verify and protect those downloads using PHP-TUF.
+ *
+ * The existence of this class is a deviation from the TUF specification. It should not be
```

Comment:
I am not sure  "deviation from the TUF specification" is accurate.

It is just that our implementation of the spec is now dependent on other parts of this library and not encapsulated in `php-tuf` itself. `deviation` would make me think we are specifically not implementing parts of the spec where i think we are just not relying on `php-tuf` to implement the complete spec and we do the parts we are not relying on `php-tuf` for.

If indeed when using this library parts of the spec are not being implemented at all we should document those parts explicitly

Reply:
Fair enough. Removed that bit.